### PR TITLE
Organizer or Group Name Feature added 

### LINKED ISSUE #4662: Create "Organizer/Group Profiles" Feature
For future features like recurring events it is important to keep the set up process simple. A feature that can help here is the "Group Profile" feature. An group page needs to provide the feature "follow group" and show events on the group page.

Please provide the option of group features as follows:

**1. Private Pages**
- [x] Add "Group" in the dropdown menu above "Manage Events" using the slug /orgs/
- [x] Show two tabs
- [x] Following /orgs/following
- [x] Organizations at /orgs/my-orgs
- [x] Show the button on top right "Create Group"
- [x] Implement the feature "Create a New Group" (which slug should that be?)
Screenshot from 2020-07-29 23-52-28

**2. Public Pages**
Provide a public "Group" page at `/[groupname]` with the following fields:
 - [x] Organizer or Group Name
 - [x] Organizer/Group Logo
 - [x] About the Organizer or Group
 - [x] Organizer/Group Image (similar to meetup on to be shown on top of organizer page)
 - [x] Website
 - [x] Social Media
 - [x] Follow
 - [ ] Contact (when clicking the contact form opens, for logged in users only)
 - [x] Show list of upcoming and past events with image and link to it
- [x] Show an "Edit Button" for org owners directing to an edit form at `/[groupname]/edit`

-  [x] **Team page**
~~Open question is how to implement the team permissions? Also in comparison to team permissions of an event?~~ Done

Screenshot from 2020-07-30 00-04-19

Related to https://github.com/fossasia/open-event-frontend/issues/5041

### PR DESCRIPTION
**Pull request by mrDarkk**
Fixes #4662

#### Short description of what this resolves:
Create "Organizer/Group Profiles" Feature added 

Screenshot 2020-08-12 at 11 49 02 AM
Screenshot 2020-08-12 at 11 49 35 AM

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 24
           

Complexity increasing per file
==============================
- app/components/org-group-form.js  1
- app/routes/org-group-profile.js  1
- tests/unit/routes/org-group-profile-test.js  1
- tests/integration/components/org-group-form-test.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753&bid=19674220)

File: app/components/org-group-form.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import Component from '@glimmer/component';
+
+import FormMixin from 'open-event-frontend/mixins/form';
+//OrgGroupForm
+export default Component.extend(FormMixin, {
+
+    getValidationRules() {
+        return {
+          inline : true,
+          delay  : false,
```
Comment (niranjan94) [inline app/components/org-group-form.js:10]: Codacy found an issue: [Expected indentation of 6 spaces but found 10.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          on     : 'blur',
+    
+          fields: {
```
Comment (niranjan94) [inline app/components/org-group-form.js:13]: Codacy found an issue: [Expected indentation of 6 spaces but found 10.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                  prompt : this.l10n.t('Please enter your Organizer or Group name')
+                }
+              ]
+            },
+            family: {
+              identifier : 'details',
+              rules      : [
+                {
+                  type   : 'empty',
+                  prompt : this.l10n.t('Please enter about the Organizer or Group')
```
Comment (niranjan94) [inline app/components/org-group-form.js:28]: Codacy found an issue: [Expected indentation of 14 spaces but found 18.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        return {
+          inline : true,
+          delay  : false,
+          on     : 'blur',
+    
+          fields: {
+            name: {
+              identifier : 'name',
+              rules      : [
+                {
```
Comment (niranjan94) [inline app/components/org-group-form.js:17]: Codacy found an issue: [Expected indentation of 12 spaces but found 16.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                  type   : 'empty',
```
Comment (niranjan94) [inline app/components/org-group-form.js:18]: Codacy found an issue: [Expected indentation of 14 spaces but found 18.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                  prompt : this.l10n.t('Please enter your Organizer or Group name')
+                }
+              ]
+            },
```
Comment (niranjan94) [inline app/components/org-group-form.js:22]: Codacy found an issue: [Expected indentation of 8 spaces but found 12.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  actions: {
+    submit(data) {
+      this.onValid(() => {
+        this.sendAction('save', data);
+      });
+    },
+  }
+});
+
+
```
Comment (niranjan94) [inline app/components/org-group-form.js:45]: Codacy found an issue: [Too many blank lines at the end of file. Max of 1 allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+            },
+            family: {
+              identifier : 'details',
+              rules      : [
+                {
+                  type   : 'empty',
+                  prompt : this.l10n.t('Please enter about the Organizer or Group')
+                }
+              ]
+            },
```
Comment (niranjan94) [inline app/components/org-group-form.js:31]: Codacy found an issue: [Unexpected trailing comma.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+          on     : 'blur',
+    
+          fields: {
+            name: {
+              identifier : 'name',
+              rules      : [
+                {
+                  type   : 'empty',
+                  prompt : this.l10n.t('Please enter your Organizer or Group name')
+                }
```
Comment (niranjan94) [inline app/components/org-group-form.js:20]: Codacy found an issue: [Expected indentation of 12 spaces but found 16.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import Component from '@glimmer/component';
+
+import FormMixin from 'open-event-frontend/mixins/form';
+//OrgGroupForm
```
Comment (niranjan94) [inline app/components/org-group-form.js:4]: Codacy found an issue: [Expected space or tab after '//' in comment.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+export default Component.extend(FormMixin, {
+
+    getValidationRules() {
```
Comment (niranjan94) [inline app/components/org-group-form.js:7]: Codacy found an issue: [Expected indentation of 2 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+            name: {
+              identifier : 'name',
+              rules      : [
+                {
+                  type   : 'empty',
+                  prompt : this.l10n.t('Please enter your Organizer or Group name')
+                }
+              ]
+            },
+            family: {
```
Comment (niranjan94) [inline app/components/org-group-form.js:23]: Codacy found an issue: [Expected indentation of 8 spaces but found 12.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+    getValidationRules() {
+        return {
+          inline : true,
+          delay  : false,
+          on     : 'blur',
+    
+          fields: {
+            name: {
+              identifier : 'name',
```
Comment (niranjan94) [inline app/components/org-group-form.js:15]: Codacy found an issue: [Expected indentation of 10 spaces but found 14.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+              rules      : [
+                {
+                  type   : 'empty',
+                  prompt : this.l10n.t('Please enter your Organizer or Group name')
+                }
+              ]
+            },
+            family: {
+              identifier : 'details',
```
Comment (niranjan94) [inline app/components/org-group-form.js:24]: Codacy found an issue: [Expected indentation of 10 spaces but found 14.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 15]
```diff
@@ -0,0 +1,45 @@
+import Component from '@glimmer/component';
+
+import FormMixin from 'open-event-frontend/mixins/form';
+//OrgGroupForm
+export default Component.extend(FormMixin, {
+
+    getValidationRules() {
+        return {
+          inline : true,
```
Comment (niranjan94) [inline app/components/org-group-form.js:9]: Codacy found an issue: [Expected indentation of 6 spaces but found 10.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+    
+          fields: {
+            name: {
+              identifier : 'name',
+              rules      : [
+                {
+                  type   : 'empty',
+                  prompt : this.l10n.t('Please enter your Organizer or Group name')
+                }
+              ]
```
Comment (niranjan94) [inline app/components/org-group-form.js:21]: Codacy found an issue: [Expected indentation of 10 spaces but found 14.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+export default Component.extend(FormMixin, {
+
+    getValidationRules() {
+        return {
+          inline : true,
+          delay  : false,
+          on     : 'blur',
+    
+          fields: {
+            name: {
```
Comment (niranjan94) [inline app/components/org-group-form.js:14]: Codacy found an issue: [Expected indentation of 8 spaces but found 12.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+              identifier : 'name',
+              rules      : [
+                {
+                  type   : 'empty',
+                  prompt : this.l10n.t('Please enter your Organizer or Group name')
```
Comment (niranjan94) [inline app/components/org-group-form.js:19]: Codacy found an issue: [Expected indentation of 14 spaces but found 18.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/routes/org-group-profile.js [hunk 19]
```diff
@@ -0,0 +1,11 @@
+import Route from '@ember/routing/route';
+
+
+
```
Comment (niranjan94) [inline app/routes/org-group-profile.js:4]: Codacy found an issue: [More than 2 blank lines not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+
+import FormMixin from 'open-event-frontend/mixins/form';
+//OrgGroupForm
+export default Component.extend(FormMixin, {
+
+    getValidationRules() {
+        return {
+          inline : true,
+          delay  : false,
+          on     : 'blur',
```
Comment (niranjan94) [inline app/components/org-group-form.js:11]: Codacy found an issue: [Expected indentation of 6 spaces but found 10.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+    
+          fields: {
+            name: {
+              identifier : 'name',
+              rules      : [
```
Comment (niranjan94) [inline app/components/org-group-form.js:16]: Codacy found an issue: [Expected indentation of 10 spaces but found 14.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+import FormMixin from 'open-event-frontend/mixins/form';
+//OrgGroupForm
+export default Component.extend(FormMixin, {
+
+    getValidationRules() {
+        return {
+          inline : true,
+          delay  : false,
+          on     : 'blur',
+
```
Comment (niranjan94) [inline app/components/org-group-form.js:12]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 23]
```diff
@@ -0,0 +1,45 @@
+import Component from '@glimmer/component';
+
+import FormMixin from 'open-event-frontend/mixins/form';
+//OrgGroupForm
+export default Component.extend(FormMixin, {
+
+    getValidationRules() {
+        return {
```
Comment (niranjan94) [inline app/components/org-group-form.js:8]: Codacy found an issue: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

File: app/components/org-group-form.js [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+              identifier : 'details',
+              rules      : [
+                {
+                  type   : 'empty',
+                  prompt : this.l10n.t('Please enter about the Organizer or Group')
+                }
+              ]
+            },
+          }
+        };
```
Comment (niranjan94) [inline app/components/org-group-form.js:33]: Codacy found an issue: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

**Comment by iamareebjamal** [comment]
How can this work when there is no server work done for this feature?

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- tests/unit/routes/org-group-profile-test.js  1
- app/components/org-group-form.js  1
- app/routes/org-group-profile.js  1
- tests/integration/components/org-group-form-test.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753&bid=19674220)

File: app/routes/org-group-profile.js (hunk 19 above)
Comment (niranjan94) [inline app/routes/org-group-profile.js:4]: Codacy found an issue: [More than 2 blank lines not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/unit/routes/org-group-profile-test.js  1
- app/components/org-group-form.js  1
- tests/integration/components/org-group-form-test.js  1
- app/routes/org-group-profile.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753&bid=19674220)

**Comment by mrDarkk** [comment]
> How can this work when there is no server work done for this feature?

its just simple ui .  its not working with server

**Comment by iamareebjamal** [comment]
Then it's not of much use until the server is ready

**Comment by mrDarkk** [comment]
> Then it's not of much use until the server is ready

okay

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/unit/routes/org-group-profile-test.js  1
- app/routes/org-group-profile.js  1
- app/components/forms/org-group-form.js  1
- tests/integration/components/org-group-form-test.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6000753&bid=19674220)

**Comment by mariobehling** [comment]
This PR is lacking server support and cannot work therefore. I am closing this issue therefore. Would be great to see this implemented on the server side as well and come back to it then.

**Pull request closed without merging**